libcaf_core/caf/unit.test.cpp

```diff
@@ -54,4 +54,11 @@ TEST("unit_results") {
   }
 }
 
+TEST("actor_address") {
+  actor_system_config cfg;
+  actor_system sys{cfg};
+  scoped_actor self{sys};
+  check(self.address().id());
```

Comment:
This doesn't return a boolean.

```suggestion
  check_ne(self.address().id(), 0);
```